Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -1179,6 +1179,12 @@ namespace AssetProcessor
                     {
                         AssetProcessor::ProcessingJobInfoBus::Broadcast(&AssetProcessor::ProcessingJobInfoBus::Events::BeginCacheFileUpdate, fullProductPath.toUtf8().data());
                         bool wasRemoved = QFile::remove(fullProductPath);
+                        if (!wasRemoved)
+                        {
+                            constexpr int DeleteRetryDelay = 10;
+                            AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(DeleteRetryDelay));
```

Comment:
This solution makes me nervous. It feels like it's just going to be hiding a problem instead of fixing it.

Reply:
This is probably copied from another part of this file where its doing the same thing (and probably should be refactored into a function).

The editor's streaming system ends up keeping a lot of cache files open with locks on them. BeginCacheFileUpdate is meant to tell the streaming system to stop it, but that message is fire-and-forget, and often doesn't get to the editor fast enough, so we're kinda left with having to retry the delete.